### PR TITLE
Pin aspire workload version

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade-services/issues/3242

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [x] Internal Infrastructure Improvements

### Release Note Description
Installs the aspire workload specified in Directory.Props

### CODE REVIEW

File: eng/templates/steps/build.yml [hunk 1]
```diff
@@ -35,10 +35,21 @@ steps:
...
+    }
+    "@
+    $fileContent | Out-File -FilePath $(Agent.TempDirectory)\aspire.json
+  displayName: Create Aspire Rollback File
+
 - powershell: |
     . .\eng\common\tools.ps1
     InitializeDotNetCli -install:$true
-    .\.dotnet\dotnet workload install aspire
+    .\.dotnet\dotnet workload install aspire --from-rollback-file $(Agent.TempDirectory)\aspire.json
```
Comment (oleksandr-didyk) [inline eng/templates/steps/build.yml:52]: Is this the only way to pin a workload version? Seems overly complicated

Comment (dkurepa) [reply eng/templates/steps/build.yml:52]: yup, `dotnet workload install` doesn't support anything else

Comment (oleksandr-didyk) [reply eng/templates/steps/build.yml:52]: It is what it is then.

Could you please include a small comment explaining why this is done? Thanks!

File: eng/templates/steps/build.yml [hunk 2]
```diff
@@ -35,10 +35,21 @@ steps:
...
+    [xml]$packagesProps = Get-Content -Path .\Directory.Packages.props
+    $aspireNode = $packagesProps.Project.ItemGroup.PackageVersion | Where-Object { $_.Include -eq 'Aspire.Hosting' }
+    $fileContent = @"
+    {
+      `"microsoft.net.sdk.aspire`": `"$($aspireNode.Version)/8.0.100`"
+    }
+    "@
+    $fileContent | Out-File -FilePath $(Agent.TempDirectory)\aspire.json
+  displayName: Create Aspire Rollback File
+
```
Comment (premun) [inline eng/templates/steps/build.yml:48]: Should this file/step not be part of the csproj? People installing this won't be able to build the project from VS?

**Comment by premun** [comment]
@dkurepa it seems like it would be quite a hassle to make devs install the right version of the workload so maybe we should just try to keep making it work with the latest?

But also pinning sounds like the right thing to do, so I don't know..

**Pull request closed without merging**